### ISSUE TITLE
pull-kubernetes-e2e-gce-etcd3 failing: Quota 'SUBNETWORKS' exceeded

### ISSUE THREAD

**Issue by cblecker** [kubernetes/kubernetes#51136]
We've bumped this quota a few times, but still seeing issues with possibly leaking resources.
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/47583/pull-kubernetes-e2e-gce-etcd3/49039/?log#log

```
I0822 18:14:36.238] Creating new network: e2e-49039
W0822 18:15:29.349] ERROR: (gcloud.compute.networks.create) Could not fetch resource:
W0822 18:15:29.350]  - Quota 'SUBNETWORKS' exceeded.  Limit: 150.0
W0822 18:15:29.350] 
W0822 18:15:29.422] 2017/08/22 18:15:29 util.go:133: Step './hack/e2e-internal/e2e-up.sh' finished in 1m45.174357334s
```

---
/kind flake
@kubernetes/sig-network-bugs @kubernetes/sig-testing-bugs @krzyzacy
Related issues:
https://github.com/kubernetes/kubernetes/issues/48688
https://github.com/kubernetes/kubernetes/issues/47362
https://github.com/kubernetes/kubernetes/issues/46713

**Comment by krzyzacy** [comment]
cc @MrHohn - any potential clue?

**Comment by MrHohn** [comment]
1\) We allow 12 pr jobs run in parallel.
2\) Each pr job has its own auto network. And we have an unused default auto network.
3\) Each auto network has 11 subnetworks (for 11 regions).
4\) We have 150 subnetworks quota.

We are running the edge -- (12 + 1) * 11 = 143 subnetworks. New pr jobs will hit quota limit if we leak any network resource.

I think we either:
1\) Need a way to bump our quota automatically when GCP expand regions.
OR 2\) Make our pr jobs framework not consuming more subnetworks when GCP expand regions.

**Comment by MrHohn** [comment]
> 1) We allow 12 pr jobs run in parallel.

https://github.com/kubernetes/test-infra/blob/e78f8fc847b9ff0625fa5382dc980ff22ab4cc83/jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-pull-json.yaml#L89

**Comment by MrHohn** [comment]
> New pr jobs will hit quota limit if we leak any network resource.

Certainly we should investigate why resource was leaked, though make sense to me to have headroom to tolerate leakage in some degree.

**Comment by krzyzacy** [comment]
@MrHohn just figured out janitor was consistently failing due to some gcloud changes. I'm going to pin it to an older version for now.

**Comment by cblecker** [comment]
@krzyzacy same issue as https://github.com/kubernetes/kubernetes/issues/49673 ?

**Comment by krzyzacy** [comment]
what I got was:
```
I0823 20:35:41.557] ['gcloud', 'compute', '-q', 'target-https-proxies', 'list', '--format=json(name,creationTimestamp.date(tz=UTC),zone,region,isManaged)', '--filter=NOT tags.items:do-not-delete AND NOT name ~ ^default', '--project=k8s-jkns-pr-gce-etcd3']
W0823 20:35:41.657] ERROR: (gcloud.compute.target-https-proxies.list) Some requests did not succeed:
W0823 20:35:41.658]  - Invalid value for field 'filter': '( tags.items ne ".*do\-not\-delete.*" ) ( name ne "^default" )'. Invalid list filter expression.
W0823 20:35:41.658] 
```

**Comment by ixdy** [comment]
@krzyzacy it looks like that filter is broken in gcloud 167.0.0 but fixed in 168.0.0.

I really wish `gcloud` would stop breaking everything.

**Comment by krzyzacy** [comment]
side notes here:
I'm getting
```
['gcloud', 'compute', '-q', 'networks', 'list', '--format=json(name,creationTimestamp.date(tz=UTC),zone,region,isManaged)', '--filter=NOT tags.items:do-not-delete AND NOT name ~ ^default', '--project=k8s-jkns-e2e-gci-gke-ci-reboot']
ERROR: (gcloud.compute.networks.list) Some requests did not succeed:
 - Invalid value for field 'filter': '( tags.items ne ".*\bdo\-not\-delete\b.*" ) ( name ne ".*(^default).*" )'. Invalid list filter expression.
```
again with gcloud 180.0.0, that's not a good sign :-(

**Comment by cblecker** [comment]
@krzyzacy yeah, something has changed with the SDK.

In 168.0.0, when I run the command (with the --log-http flag), it actually gets all the networks in the project, but then filters them client-side. When I do the same in 180.0.0, it actually puts the filter as part of the query, and it's run by the upstream API. This makes sense, but yeah, certain filter queries aren't working when they should. I don't see tags as part of the network object API, but even if I replace that with something like autoCreateSubnetworks, I still get an invalid expression error.

example command: `gcloud compute networks list --filter='NOT autoCreateSubnetworks=True AND name ~ ^default' --log-http`
example call it makes: `GET /compute/v1/projects/<project_name>/global/networks?filter=%28+autoCreateSubnetworks+ne+true+%29+%28+name+eq+%22.%2A%28%5Edefault%29.%2A%22+%29&alt=json&maxResults=500 HTTP/1.1`
example response:
```json
{
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "invalid",
    "message": "Invalid value for field 'filter': '( autoCreateSubnetworks ne true ) ( name eq \".*(^default).*\" )'. Invalid list filter expression."
   }
  ],
  "code": 400,
  "message": "Invalid value for field 'filter': '( autoCreateSubnetworks ne true ) ( name eq \".*(^default).*\" )'. Invalid list filter expression."
 }
}
```
